python/CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+# =============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+cmake_minimum_required(VERSION 3.20.1 FATAL_ERROR)
+
+include(../fetch_rapids.cmake)
+
+include(rapids-cmake)
+include(rapids-cpm)
+include(rapids-find)
+
+set(CUML_VERSION 22.08.00)
+
+project(
+  cuml-python
+  VERSION ${CUML_VERSION}
+  LANGUAGES # TODO: Building Python extension modules via the python_extension_module requires the C
+            # language to be enabled here. The test project that is built in scikit-build to verify
+            # various linking options for the python library is hardcoded to build with C, so until
+            # that is fixed we need to keep C.
+            C CXX
+)
+
+################################################################################
+# - User Options  --------------------------------------------------------------
+option(FIND_CUML_CPP "Search for existing CUML C++ installations before defaulting to local files" OFF)
+option(SINGLEGPU "Disable all mnmg components and comms libraries" OFF)
+
+message(VERBOSE "CUML_PY: Searching for existing CUML C++ installations before defaulting to local files: ${FIND_CUML_CPP}")
+message(VERBOSE "CUML_PY: Disabling all mnmg components and comms libraries: ${SINGLEGPU}")
+
+set(CUML_CPP_TARGET "cuml++")
+
+################################################################################
+# - Process User Options  ------------------------------------------------------
+
+
+# If the user requested it,  we attempt to find cuml.
+if(FIND_CUML_CPP)
+  rapids_cpm_init()
+  include(rapids-cuda)
+  rapids_cuda_init_architectures(cuml-python)
+  enable_language(CUDA)
+
+  set(CUML_VERSION_MAJOR "22")
+  set(CUML_VERSION_MINOR "08")
+  set(CUML_USE_RAFT_NN ON)
+  set(CUML_USE_RAFT_DIST ON)
+
+  # We need to call get_raft due to cuML asing for raft::nn and
+  # raft::distance targets
+  include(../cpp/cmake/thirdparty/get_raft.cmake)
```

Comment:
Shouldn't the raft targets be exported from the libcuml target? I see that [they're part of the public link interface](https://github.com/rapidsai/cuml/blob/branch-22.08/cpp/CMakeLists.txt#L519) and [raft::raft is part of the export sets](https://github.com/rapidsai/cuml/blob/branch-22.08/cpp/cmake/thirdparty/get_raft.cmake#L56). Should `get_raft.cmake` be using `PKG_USE_RAFT_DIST`, `PKG_USE_RAFT_NN`, and/or `RAFT_COMPILE_LIBRARIES` to decide whether or not to manually add `raft::nn` and/or `raft::distance` to the export sets as well? CC @robertmaynard

Reply:
Actually thinking about it for a bit, I think we just need to add `raft::nn` and `raft::distance` to https://github.com/rapidsai/cuml/blob/5324d52801fc63e5fd5eced47045b7bb95a17a12/cpp/cmake/thirdparty/get_raft.cmake#L55 should be enough to properly add them to libcuml's exported targets, will try it locally

Reply:
I think that the raft::nn and raft::distance are made GLOBAL when they are brought in as well. ( if not they should be )

Reply:
They are https://github.com/rapidsai/raft/blob/branch-22.08/cpp/CMakeLists.txt#L493

Reply:
It seems then there is a bug somewhere, without calling get_raft.cmake, or modifying get_raft.cmake in cuML, we run into the following issue: 

```bash
CMake Error at /home/ursa/miniconda3/envs/skbuild4/lib/cmake/cuml/cuml-targets.cmake:56 (set_target_properties):
  The link interface of target "cuml::cuml++" contains:

    raft::nn

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /home/ursa/miniconda3/envs/skbuild4/lib/cmake/cuml/cuml-config.cmake:77 (include)
  CMakeLists.txt:68 (find_package)
```

Reply:
The dependency tracking in rapids-CMake doesn't understand COMPONENTs and therefore cuml is find raft but not enabling any components. 
I think we can re call find package raft with the right components to unblock us. 

Please file an issue on rapids-CMake about this